packages/core/src/database/index.ts

```diff
@@ -985,84 +811,88 @@ EXECUTE PROCEDURE "${namespace}".${notification};`),
                 ),
               );
 
-              await createEnums();
-              await createTables();
-              createdTables = true;
-            }
+              await createEnums(tx);
+              await createTables(tx);
 
-            if (createdTables) {
               common.logger.info({
                 service: "database",
                 msg: `Created tables [${tables.map(getTableName).join(", ")}]`,
               });
 
-              // write metadata
+              await tx.update(PONDER_META).set({ value: metadata });
+              return {
+                status: "success",
+                crashRecoveryCheckpoint: undefined,
+              } as const;
+            }
+
+            if (
+              common.options.command === "dev" ||
```

Comment:
Should this be an and?

Reply:
No, this is logically what I intended. 

An error is thrown: 
1. if a `dev` deployment` is trying to replace a `start` deployment
2. if a `start` deployment is trying to replace a `start` deployment but `buildId` is different